compiler/kernel/op.go

```diff
@@ -244,6 +247,94 @@ func (b *Builder) compileLeaf(o dag.Op, parent zbuf.Puller) (zbuf.Puller, error)
 			return nil, err
 		}
 		return b.compileOver(parent, v.Over, names, exprs)
+	case *dag.PoolScan:
+		if parent != nil {
+			return nil, errors.New("a pool scan cannot have a parent")
```

Comment:
Nit: Make this consistent with the dag.Lister error.
```suggestion
			return nil, errors.New("pool scan cannot have a parent operator")
```